src/ui/page/dataverse/dataset/dataset.tsx

```diff
@@ -91,15 +91,29 @@ const datasetMetadata: ItemGeneralMetadata[] = [
   }
 ]
 
-const Dataset = (): JSX.Element => {
+const isDataSet = (resource: DataverseItemDetails): resource is Dataset =>
+  resource.type === 'dataset'
+
+const Dataset: FC = () => {
   const { id } = useParams<string>()
-  const [dataset, setDataset] = useState<Option<DataverseItemDetails>>(none)
+  const [dataset, setDataset] = useState<O.Option<DataverseItemDetails>>(O.none)
+  const [isLoading, setIsLoading] = useState(true)
 
   useEffect(() => {
-    id && setDataset(getResourceDetails(id))
+    setIsLoading(true)
```

Comment:
I think you don't need to set again the loading to true because you already made it into useState initialization ;)

Reply:
You are right, React reconciles with a different type. Good catch, thanks!